### PR TITLE
net: http: Improve HTTP_USER_AGENT version number

### PR DESCRIPTION
**Pull request by miyatsu**
Signed-off-by: Ding Tao <miyatsu@qq.com>

### CODE REVIEW

File: include/net/http.h [hunk 1]
```diff
@@ -130,8 +130,12 @@ enum http_final_call {
 #endif
 
 #ifndef HTTP_USER_AGENT
-#define HTTP_USER_AGENT	"Zephyr-HTTP-Client/1.8"
-#endif
+#if defined(CONFIG_HTTPS)
+#define HTTP_USER_AGENT	"Zephyr-HTTPS-Client/"KERNEL_VERSION_STRING
```
Comment (pfalcon) [inline include/net/http.h]: I'd suggest to put space after closing `"` here and below.

**Comment by pfalcon** [comment]
Looks good, please squash 2 commits into 1.

**Comment by lpereira** [comment]
Why there should be a distinction between HTTP and HTTPS in the user agent header?

**Review by lpereira — CHANGES_REQUESTED** [review]
Why there should be a distinction between HTTP and HTTPS in the user agent header?  It's still the same protocol; just the transport that changes.

In fact, if we were to improve this header, I wouldn't send out the Zephyr version number at all: the user-agent header is nearly useless, and sending the kernel version number with every request will send out information that can be used to determine if a device is using a (potentially) vulnerable version of Zephyr or not.

**Comment by miyatsu** [comment]
Well, I'm just wondering how the device gonna do while received a 302 redirection to a HTTPs URL.

The web server can check the user agent head field, to determine which URL (HTTP or HTTPs) should response with status 302.

I've read the code of http_client.c, it seems the device did not support for 302 redirection auto connect to the URL within Location field. Using the kernel version at user agent head may useful for web server to determine witch features are supported by the device, 302 redirection for example.

**Comment by lpereira** [comment]
This won't distinguish between the HTTPS or the HTTP version on a redirect. It's always going to use the HTTPS one if TLS is built in, and the HTTP one if not.  The server should perform the redirect based on the port the client connected to (80 vs 443), not the user agent.  Most of the time, the user agent string is useless and should be ignored by the server.

**Pull request closed without merging**